lib/my_inject_rec.rb

```diff
@@ -0,0 +1,35 @@
+class Array
+  def my_inject_rec(*args, &block)
+    start, accumulator = start_and_accumulator_rec(args)
+    if block
+      accumulator = run_block_rec(start, accumulator, block)
+    else
+      accumulator = run_symbol_rec(start, accumulator, args[-1])
+    end
+    accumulator
+  end
+
```

Comment:
See before for use of `private` - at the moment any object can access `run_block_rec()` etc